hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/ApacheHttpClientMessageBatchSender.java

```diff
@@ -64,6 +66,11 @@ public MessageSendingResult send(MessageBatch batch, URI address, int requestTim
         httpPost.addHeader(HTTP.CONTENT_TYPE, contentType.getMimeType());
         httpPost.addHeader(RETRY_COUNT.getName(), Integer.toString(batch.getRetryCounter()));
 
+        if (batch.hasSubscriptionIdentityHeaders()) {
```

Comment:
Do we need to check this condition every time during sending message ? You can check condition and add required headers during subscription initialization only once :) This way you always would just iterate over headers. WDYT ?